linker/shared/src/main/scala/org/scalajs/linker/frontend/optimizer/OptimizerCore.scala

```diff
@@ -5280,4 +5316,12 @@ private[optimizer] object OptimizerCore {
         private[FreshNameAllocator] val usedNamesToNextCounter: Map[String, Int])
   }
 
+  type FieldID = String
+
+  def fieldID(field: Ident): FieldID =
+    field.name
```

Comment:
Making it an `AnyVal` is going to allocate more and cause more indirections, because it's used quite a lot in type parameters (of `Map`s, notably), so I wouldn't do that. I'd like to make it an `opaque type`, but that doesn't exist yet ;)

I'll make sure to always use appropriate constructors, though.

Reply:
Well, yes, for two commits. And then you replace it with a tuple anyways. So the `AnyVal` would become a case class and be on par, no?